edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/RoutingEdgeHub.cs

```diff
@@ -387,6 +396,36 @@ public static void ProcessingSubscription(string id, DeviceSubscription deviceSu
             {
                 Log.LogInformation((int)EventIds.ProcessingSubscription, Invariant($"Processing subscription {deviceSubscription} for client {id}."));
             }
+
+            public static void MessagesReceived(IIdentity identity, IList<IMessage> messages)
+            {
+                if (messages.Count > 1)
+                {
+                    IEnumerable<string> messageIds = messages
```

Comment:
Should we check if debug logs are enabled before going through the work of generating this ids string?

Reply:
Yup, fixed.